src/StreamContextFactory.php

```diff
@@ -30,13 +31,29 @@ public function create(ConfigInterface $config)
             if (isset($proxy['proxy_login']) && isset($proxy['proxy_password'])) {
                 // Support for proxy authentication is untested. The current implementation is based on
                 // http://php.net/manual/en/function.stream-context-create.php#74431.
-                $options['http']['header'] =  array(
-                    'Proxy-Authorization: Basic ' .
-                    base64_encode($proxy['proxy_login'] . ':' . $proxy['proxy_password'])
-                );
+                $headers[] = 'Proxy-Authorization: Basic ' .
+                    base64_encode($proxy['proxy_login'] . ':' . $proxy['proxy_password']);
             }
         }
 
+        // Native SoapClient uses basic authentication by defaults,
+        // and `authentication` option can be omitted.
+        $soapOptions = $config->get('soapClientOptions') + array(
+            'authentication' => SOAP_AUTHENTICATION_BASIC
+        );
```

Comment:
Sorry, my english is lame.

I've added merging arrays to ensure that the `$soapOptions` contains `authentication` key. This is allow to simplify the condition. Otherwise we would have to write condition, like this:

```php
if ((!isset($soapOptions['authentication']) || $soapOptions['authentication'] === SOAP_AUTHENTICATION_BASIC) &&
    isset($soapOptions['login']) &&
    isset($soapOptions['password'])
) {
```

I can drop merging arrays and rewrite the condition.